src/codecs/avif/options.tsx

```diff
@@ -1,72 +1,304 @@
 import { h, Component } from 'preact';
-import { bind } from '../../lib/initial-util';
-import {
-  /*inputFieldCheckedAsNumber,*/ inputFieldValueAsNumber,
-  preventDefault,
-} from '../../lib/util';
-import { EncodeOptions } from './encoder-meta';
+import { preventDefault, shallowEqual } from '../../lib/util';
+import { EncodeOptions, defaultOptions } from './encoder-meta';
 import * as style from '../../components/Options/style.scss';
-// import Checkbox from '../../components/checkbox';
-// import Expander from '../../components/expander';
-// import Select from '../../components/select';
+import Checkbox from '../../components/checkbox';
+import Expander from '../../components/expander';
+import Select from '../../components/select';
 import Range from '../../components/range';
-// import linkState from 'linkstate';
+import linkState from 'linkstate';
 
 interface Props {
   options: EncodeOptions;
   onChange(newOptions: EncodeOptions): void;
 }
 
-interface State {}
+interface State {
+  options: EncodeOptions;
+  lossless: boolean;
+  maxQuality: number;
+  minQuality: number;
+  separateAlpha: boolean;
+  losslessAlpha: boolean;
+  maxAlphaQuality: number;
+  minAlphaQuality: number;
+  showAdvanced: boolean;
+  grayscale: boolean;
+  subsample: number;
+  tileRows: number;
+  tileCols: number;
+  effort: number;
+}
```

Comment:
I went in a slightly different direction with these options compared to others.

Other options try to set the form state from the incoming encoder options, but that gets really complicated in cases where the form represents things slightly differently. For example, here we have 'lossless' as a toggle, but toggling that shouldn't change the quality slider, just the output values. Also, if toggling one option results in a form field being removed, code has to be defensive about that.

Instead, I've kept the form state and encoder options separate. Encoder options are converted to state whenever they've changed from the outside, in `getDerivedStateFromProps`. Form options are converted to encoder options in `_inputChange`.

@developit is this a reasonable use of `getDerivedStateFromProps`?